sites/all/modules/custom/gm3/gm3_field/gm3_field.module

```diff
@@ -754,10 +755,8 @@ function gm3_field_field_is_empty($item, $field){
             }
           }
         }else{
-          $non_empty = array_filter($item, function ($i){
-            return $i !== NULL;
-          });
-          return !count($non_empty);
+          $gm3_type = $item['gm3_type'];
+          return empty($item[$gm3_type]);
```

Comment:
I think this won't work if `gm3_type` is `point`; in that case you need to see if `latitude` or `longitude` is empty